default/scripting/buildings/TERRAFORM.focs.txt

```diff
@@ -84,5 +84,80 @@ TARGET_TERRAFORMING
     icon = "icons/building/terraform.png"
 '''
 
+BuildingType
+    name = "BLD_TERRAFORM_BEST"
+    description = "BLD_TERRAFORM_BEST_DESC"
+    buildcost =  75 * (1 - 0.4 * Statistic If condition = And [Source EmpireHasAdoptedPolicy empire = Source.Owner name = "PLC_TERRAFORMING"])
```

Comment:
This one costs only 75% (and 2/3 time) of the targeted version. This is to avoid small wins by "micro-managment"? I think 90% would do for that.

Reply:
there are real wins for micromanaging multisteps, because the population can grow much sooner

Reply:
one could probably implement multisteps by having an effectsgroup checking if the building is enqueued and how far progress is. if somebody does that, we can go to 90% or maybe even remove
i am not sure how much shortening the buildtime lessens the problem; i would also accept a count-of-step-based buildcost (e.g. 0% discount for 1 step, 10% discount for 2 steps, 21% for 3, 33% for 4, 46% for 5...)

Reply:
so id say 10% linear discount, it should only maximum 4 steps anyway.

Reply:
ok, cant do anything with floats there, it seems the parser is broken round(xxx) is not excepted as integer expression

Reply:
not broken, rather a missing feature